src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/application/actions/DrawGacha.scala

```diff
@@ -0,0 +1,15 @@
+package com.github.unchama.seichiassist.subsystems.gacha.application.actions
+
+import com.github.unchama.seichiassist.subsystems.gacha.GachaAPI
+
+trait DrawGacha[F[_], Player] {
+
+  def draw(player: Player, amount: Int)(implicit gachaAPI: GachaAPI[F]): F[Unit]
```

Comment:
```suggestion
  def draw(player: Player, amount: Int): F[Unit]
```

API をインターフェースで要求せずに、この trait の実装が `GachaAPI[F]` を自分で持ってくるようにしたほうが良いです